src/routes/settings/+page.svelte

```diff
@@ -80,64 +84,85 @@
 	}
 
 	async function updatePassword(event) {
-		loading = true;
+	loading = true;
+	const user = get(currentUser);
+		if (!user) {
+			throw new Error('Current user not available');
+		}
 
-		if ($currentUser) throw new Error('Current user not available');
+  	let passwordFormData;
 
 		try {
-			const userId = user?.id;
+			const userId = user.id;
 
 			const formData = new FormData(event.target);
 			const data = Object.fromEntries(formData);
 
 			const userRecord = await pb.collection('users').getFirstListItem(`id="${userId}"`);
 
-			if (data.password !== data.passwordConfirm) {
-				throw new Error('New Password & Confirm Password must match!');
-			}
+		if (!data.password || !data.passwordConfirm) {
+		throw new Error('Both password and password confirm fields are required');
+		}
+
+		if (data.password !== data.passwordConfirm) {
+		throw new Error('New Password & Confirm Password must match!');
+		}
 
+    // Verify the old password
+	try {
 			await pb.collection('users').authWithPassword(userRecord.email, data.oldPassword);
+			} catch (error) {
+			if (error.status === 401) {
+				throw new Error('Old Password Incorrect!');
+			} else {
+				throw error;
+			}
+		}
 
-			await pb.collection('users').update(userId, {
-				oldPassword: data.oldPassword,
-				password: data.password,
-				passwordConfirm: data.passwordConfirm
-			});
+			// Update the password
+				passwordFormData = new FormData(); // Initialize the variable here
+				passwordFormData.append('password', data.password);
```

Comment:
Review using this method. It's lengthy plus the object is used only in one place. Also, passFormData = undefined at first when it's initially declared. Proper declaration should be set as well. e.g., since passFormData is an object it should be initially assigned an empty object.

Note: If it's used in 3 or more times, then store it in a variable, if not then just type the object inside.